host/sysfs/gpio.go

```diff
@@ -167,19 +167,23 @@ func (p *Pin) In(pull gpio.Pull, edge gpio.Edge) error {
 				return p.wrap(err)
 			}
 		}
-		if p.edge != edge {
-			var b []byte
-			switch edge {
-			case gpio.RisingEdge:
-				b = bRising
-			case gpio.FallingEdge:
-				b = bFalling
-			case gpio.BothEdges:
-				b = bBoth
-			}
-			if err := seekWrite(p.fEdge, b); err != nil {
-				return p.wrap(err)
-			}
+		// Always reset the edge detection mode to none after starting the epoll
+		// otherwise edges are not always delivered, as observed on an Allwinner A20
+		// running kernel 4.14.14.
+		if err := seekWrite(p.fEdge, bNone); err != nil {
+			return p.wrap(err)
+		}
+		var b []byte
+		switch edge {
+		case gpio.RisingEdge:
+			b = bRising
+		case gpio.FallingEdge:
+			b = bFalling
+		case gpio.BothEdges:
+			b = bBoth
+		}
+		if err := seekWrite(p.fEdge, b); err != nil {
+			return p.wrap(err)
 		}
 	}
```

Comment:
One thing I realized is that when going from edge to no edge it's not disabling it. It's not a big deal but that's generating unneeded interrupts. That's out of scope for this change though.